plugin-server/src/main/ingestion-queues/scheduled-tasks-consumer.ts

```diff
@@ -0,0 +1,114 @@
+import Piscina from '@posthog/piscina'
+import { StatsD } from 'hot-shots'
+import { EachBatchHandler, Kafka, Producer } from 'kafkajs'
+
+import { KAFKA_SCHEDULED_TASKS, KAFKA_SCHEDULED_TASKS_DLQ } from '../../config/kafka-topics'
+import { DependencyUnavailableError } from '../../utils/db/error'
+import { status } from '../../utils/status'
+import { instrumentEachBatch, setupEventHandlers } from './kafka-queue'
+
+// The valid task types that can be scheduled.
+// TODO: not sure if there is another place that defines these but it would be
+// good to unify.
+const taskTypes = ['runEveryMinute', 'runEveryHour', 'runEveryDay'] as const
+
+export const startScheduledTasksConsumer = async ({
+    kafka,
+    piscina,
+    producer,
+    statsd,
+}: {
+    kafka: Kafka
+    piscina: Piscina
+    producer: Producer // NOTE: not using KafkaProducerWrapper here to avoid buffering logic
+    statsd?: StatsD
+}) => {
+    /*
+        Consumes from the scheduled tasks topic, and executes them within a
+        Piscina worker.
+    */
+
+    const consumer = kafka.consumer({ groupId: 'scheduled-tasks-runner' })
+    setupEventHandlers(consumer)
+
+    status.info('🔁', 'Starting scheduled tasks consumer')
+
+    const eachBatch: EachBatchHandler = async ({ batch, resolveOffset, heartbeat }) => {
+        status.debug('🔁', 'Processing batch', { size: batch.messages.length })
+        for (const message of batch.messages) {
+            if (!message.value) {
+                status.error('⚠️', 'asdf', { topic: 'zxcv' })
+                status.warn('⚠️', `Invalid message for partition ${batch.partition} offset ${message.offset}.`, {
+                    value: message.value,
+                })
+                await producer.send({ topic: KAFKA_SCHEDULED_TASKS_DLQ, messages: [message] })
+                resolveOffset(message.offset)
+                continue
+            }
+
+            let task: {
+                taskType: typeof taskTypes[number]
```

Comment:
this feels like a weird way to do things

Reply:
Do you have a suggested improvement here? Which bit is weird? I'm not so hot on Typescript 🤔

Reply:
I think we already have a type that's equivalent to `'runEveryMinute' | 'runEveryHour' | 'runEveryDay'` - I'd just use that

Reply:
Ah maybe it's in the plugin-scaffolding, I didn't check there.

One reason I did it like this was due to also using the `taskTypes` array for runtime schema validation further down as well.